source/extensions/common/wasm/wasm.h

```diff
@@ -309,6 +338,7 @@ class Wasm : public Envoy::Server::Wasm,
   std::function<void(Common::Wasm::Context*)> tick_;
   std::chrono::milliseconds tick_period_;
   Event::TimerPtr timer_;
+  Stats::ScopeSharedPtr owned_scope_;  // When scope_ is not owned by a higher level (e.g. for WASM services).
```

Comment:
This is confusing. Could you elaborate? Looks like Wasm could either own scope_ or not own. If so, we doesn't `Wasm` always have a reference to `Stats::Scope& scope_`?

Reply:
Yes, it always has a scope_.  It only has the owned_scope_ when it is not using the scope provided by the stream e.g. in the case of Filters.